processing_chain/src/test/groovy/org/orbisgis/processingchain/ProcessingChainTest.groovy

```diff
@@ -557,41 +599,125 @@ class ProcessingChainTest {
+                          String hydrographicTableName, boolean saveResults ) {
+
+        // Create the RSU
+        IProcess prepareRSUData = Geoclimate.SpatialUnits.prepareRSUData()
+        IProcess createRSU = Geoclimate.SpatialUnits.createRSU()
+        prepareRSUData.execute([datasource          : datasource,           zoneTable           : zoneTableName,
+                                roadTable           : roadTableName,        railTable           : railTableName,
+                                vegetationTable     : vegetationTableName,  hydrographicTable   : hydrographicTableName,
+                                surface_vegetation  : 100000,               surface_hydro       : 2500,
+                                prefixName          : ""])
+        createRSU.execute([datasource: datasource, inputTableName : prepareRSUData.results.outputTableName,
+                           prefixName: ""])
+        String createdRsu = createRSU.results.outputTableName
+
+
+        // Create the relations between buildings and RSU
+        IProcess createScalesRelationsRsuBu = Geoclimate.SpatialUnits.createScalesRelations()
+        createScalesRelationsRsuBu.execute([datasource                : datasource,
+                                            inputLowerScaleTableName  : buildingTableName,
+                                            inputUpperScaleTableName  : createdRsu,
+                                            idColumnUp                : createRSU.results.outputIdRsu,
+                                            prefixName                : "test"])
+
+        String relationBuildingsRSU = createScalesRelationsRsuBu.results.outputTableName
+
+        // Calculate the LCZ indicators and the corresponding LCZ class of each RSU
+        IProcess pm_lcz =  ProcessingChain.BuildLCZ.createLCZ()
+        pm_lcz.execute([datasource: datasource, prefixName: "test", buildingTable: relationBuildingsRSU,
+                        rsuTable: createdRsu, roadTable: roadTableName, vegetationTable: vegetationTableName,
+                        hydrographicTable: hydrographicTableName, facadeDensListLayersBottom: [0, 50, 200], facadeDensNumberOfDirection: 8,
+                        svfPointDensity: 0.008, svfRayLength: 100, svfNumberOfDirection: 60,
+                        heightColumnName: "height_roof", fractionTypePervious: ["low_vegetation", "water"],
+                        fractionTypeImpervious: ["road"], inputFields: ["id_build"], levelForRoads: [0]])
+        String lczResults = pm_lcz.results.outputTableName
+
+        // Check if we have the same number of RSU
+        def countRelationRSU= datasource.firstRow("select count(*) as count from $createdRsu".toString())
+        def countRSULcz = datasource.firstRow("select count(*) as count from $lczResults".toString())
+        assertEquals(countRelationRSU.count,countRSULcz.count)
+
+        if (saveResults) {
+            println("Saving LCZ classes")
```

Comment:
Use `logger.info()` instead of `println()`

Reply:
OK but logger is not defined in the test file, how do I import it ? Should I define it as in the ProcessingChain file ?
```groovy
public static Logger logger = LoggerFactory.getLogger(ProcessingChain.class)
```

Reply:
yes, juste replace the class name so it becomes :
``` groovy
public static Logger logger = LoggerFactory.getLogger(ProcessingChainTest.class)
```

Reply:
OK I have added this line in the test file right after
```groovy
class ProcessingChainTest {
private final static File bdTopoDb = new File("./target/myh2gisbdtopodb.mv.db")
```
Is it the right location ?

Reply:
yes !